README.md

```diff
@@ -0,0 +1,114 @@
+# xtm - a library for cross-thread messaging with event loop intergation
+
+The library provides the following facilities:
+
+# xtm_queue
+
+Opaque struct, that represent one-directional, one-reader-one-writer queue
+implementation with eventloop integration ability.
+
+## xtm_create
+
+Allocation function, used to create instance of struct xtm_queue and return
+its pointer or NULL with the following possible errno: ENOMEM, EINVAL, EMFILE,
+ENFILE, ENODEV. Accepts the queue size as input, which must be a power of two.
+
+## xtm_delete
+
+Deallocation function, used to free queue and close its internal fds.
+Returns 0 on success. Otherwise -1 with errno set appropriately (as in
+close(2), since it implies close of fds)
+
+## xtm_msg_notify
+
+Function for queue consumer notification.
+Returns 0 on success. Otherwise -1 with errno set appropriately (as in
+write(2), since it implies write to internal fd)
+
+## xtm_msg_probe
+
+Returns 0 if queue has space. Otherwise -1 with errno set to ENOBUFS.
+
+## xtm_msg_count
+
+Function returns current message count in queue.
+
+## xtm_fun_dispatch
```

Comment:
How is dispatch different from send? From the name it is super unclear. Moreover, dispatch name looks like it handles the elements, not pushes them.

Reply:
I talked with @Mons about it. He believes that the purpose of the functions can be understood from READMY.md. But if you have you suggestion to make it more clear, he is ready to consider!

Reply:
The zero difference between send/dispatch is still actual.

Reply:
I can't rename functions until @Mons  approves it!

Reply:
Mons is the customer, API is a part of task.

Reply:
Customers can't affect our code style. This is not a justification. API is a part of the task, but I don't see how exact names are a part of it.